sound/soc/sof/intel/hda-ctrl.c

```diff
@@ -177,9 +177,14 @@ int hda_dsp_ctrl_init_chip(struct snd_sof_dev *sdev, bool full_reset)
 
 	if (full_reset) {
 		/* clear WAKESTS */
-		snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR, SOF_HDA_WAKESTS,
-					SOF_HDA_WAKESTS_INT_MASK,
-					SOF_HDA_WAKESTS_INT_MASK);
+		snd_sof_dsp_write(sdev, HDA_DSP_HDA_BAR, SOF_HDA_WAKESTS,
+				  SOF_HDA_WAKESTS_INT_MASK);
+
+		/*
+		 * set codec_mask to 0, and get it again after reset. This
+		 * will handle the case that codecs changed during suspend.
+		 */
+		bus->codec_mask = 0;
```

Comment:
This codec_mask is reset by codec driver in hdac_hda.c and hdmi driver in hdac_hdmi.c after it is initialized by SOF.
According to legacy hda driver, these operations are not reasonable in ASoC.

This patch is a workaround for above case. I recommend doesn't change it now. 
codec_mask is only for skl_hda_generic driver to initialize dai link when PCI probed, and it is not used later.
So driver works without this patch.

Reply:
@RanderWang @keyonjie @ranj063 So we are back to the beginning.
Just remove the first clear WAKESTS, and do nothing else in the last commit.

Reply:
@zhuyingjiang I'm confused about  your comment and the commit. Do you have to set codec_mask to 0 here?

Reply:
@zhuyingjiang Maybe better to leave this out. I agree with @ranj063 that the comment+commit are not clear. Unless there's a good reason, I'd keep the current logic where we read codec_mask once after probe with:
```
»       if (!bus->codec_mask) {
»       »       bus->codec_mask = snd_hdac_chip_readw(bus, STATESTS);
```

... and not reset unless probe is done again (by unloading and reloading the modules).

Reply:
@kv2019i That is what I say we are back to the beginning. So I will push the old version of my patch.

Why set codec_mask to 0 is because I see an issue that with a printk, I see codec_mask is 0x5 at cold start, and after suspend/resume, it change to 0x1. I set codec_mask to 0 to make the codec_mask to be read again from the STATESTS, (will got 0x5).

But as @RanderWang suggested 
> This patch is a workaround

we need to fix the above issue elsewhere, maybe i915 issue or somewhere else.

And what @keyonjie suggested:
> should remove the later one, and keep this and move it out of if (full_reset), above.

As I tested on WHL RVP,  the later one can't be removed, or the device can't suspend after the first suspend.

In conclude: now we are aligned that we only need remove the first 'remove STATESTS' , that was what I commit in the previous version.

I will push the previous version again.